### PR TITLE
#5949 - Update EditNetSpell to respect comment line limits

### PR DESCRIPTION
**Pull request by lukevp**
- Update EditNetSpell to respect the line limits configured in the commit settings rather than highlighting based on hardcoded values.

Fixes #

## Proposed changes

- Update the spell check code to respect line length settings in appsettings rather than being hardcoded

### Before

image
image

### After

No highlighting with settings disabled
image

With settings enabled, highlights as before.
image
image

## Test methodology 

- Tested with settings enabled and disabled and ensured proper functionality.

## Test environment(s) 

- GIT 2.22.0.windows.1
- Windows 10 Pro 1903 Build 18362.53

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by Bejoscha** [comment]
I like this change and am all for it.

**Comment by pmiossec** [comment]
I don't like all the consequences of this change (if I understand well what it is doing)...

You are linking what is a hard commit validation feature (that must be respected) to the one of ill commit highlight (which is just a help).

So, with your changes, I can't have yellow highlight anymore (that I want to enforce good practices), without the hard limit (because I want sometimes to write longer lines and so exceed the limits).

But I like the idea to be able to change the values. But I tend to think that it should be 2 other fields that should be added.

And in the same time, perhaps also add the checkbox to enable the 'mark ill commit message' here.

**Comment by vbjay** [comment]
Read up on 50/72 rule.

File: GitUI/SpellChecker/EditNetSpell.cs [hunk 1]
```diff
@@ -432,21 +432,38 @@ private void MarkLines()
                 return;
             }
 
+            int limit1 = AppSettings.CommitValidationMaxCntCharsFirstLine;
+            int limitX = AppSettings.CommitValidationMaxCntCharsPerLine;
+            bool empty2 = AppSettings.CommitValidationSecondLineMustBeEmpty;
```
Comment (RussKie) [inline GitUI/SpellChecker/EditNetSpell.cs]: Please rename variable to better communicate their purpose, e.g.
```
limitMaxCharsFirstLine
limitMaxCharsOtherLines
secondLineMustBeEmpty
```

**Comment by RussKie** [comment]
I would like to have tests for the change.
Do you think you can add some? Happy to help you with the setup

**Comment by RussKie** [comment]
Also please update the commit message with "Fixes #5949", this way it stays in the history.

**Comment by GintasS** [comment]
This is an old PR and should be closed or set to a draft. @RussKie

**Pull request closed without merging**